File: local-movie-info-provider/src/main/java/com/github/rahmnathan/localmovies/info/provider/control/JSONtoMovieMapper.java
package com.github.rahmnathan.localmovies.info.provider.control;

import com.github.rahmnathan.localmovies.info.provider.data.Movie;

import java.util.ArrayList;
import java.util.List;

import org.json.JSONArray;
import org.json.JSONObject;

class JSONtoMovieMapper {

    static List<Movie> jsonArrayToMovieInfoList(JSONArray jsonList) {
        List<Movie> movieList = new ArrayList<>();
        for (int i = 0; i < jsonList.length(); i++) {
            JSONObject mediaFile = jsonList.getJSONObject(i);
            Movie.Builder builder = Movie.Builder.newInstance()
                    .setCreated(mediaFile.getLong("dateCreated"))
                    .setFileName(mediaFile.getString("fileName"))
                    .setViews(mediaFile.getInt("views"));

            JSONObject movieInfo = mediaFile.getJSONObject("movieInfo");

            builder.setReleaseYear(movieInfo.getString("releaseYear"))
                    .setMetaRating(movieInfo.getString("metaRating"))
                    .setIMDBRating(movieInfo.getString("imdbrating"))
                    .setTitle(movieInfo.getString("title"))
                    .setGenre(movieInfo.getString("genre"))
                    .setImage(movieInfo.getString("image"));

            movieList.add(builder.build());
        }
        return movieList;
    }
}

File: local-movie-info-provider/src/main/java/com/github/rahmnathan/localmovies/info/provider/control/JSONtoMovieMapper.kt
package com.github.rahmnathan.localmovies.info.provider.control

import com.github.rahmnathan.localmovies.info.provider.data.Movie
import com.github.rahmnathan.localmovies.info.provider.data.MovieEvent

import java.util.ArrayList

import org.json.JSONArray
import org.json.JSONObject

internal object JSONtoMovieMapper {

    fun jsonArrayToMovieInfoList(jsonList: JSONArray): List<Movie> {
        val movieList = ArrayList<Movie>()
        for (i in 0 until jsonList.length()) {
            val mediaFile = jsonList.getJSONObject(i)
            movieList.add(mediaFileToMovie(mediaFile))
        }
        return movieList
    }

    fun mediaFileToMovie(mediaFile: JSONObject): Movie {
        val movieInfo = mediaFile.getJSONObject("movie")

        return Movie(
                releaseYear = movieInfo.getString("releaseYear"),
                metaRating = movieInfo.getString("metaRating"),
                imdbRating = movieInfo.getString("imdbrating"),
                created = mediaFile.getLong("dateCreated"),
                filename = mediaFile.getString("fileName"),
                title = movieInfo.getString("title"),
                genre = movieInfo.getString("genre"),
                image = movieInfo.getString("image"),
                views = mediaFile.getInt("views"),
                actors = movieInfo.getString("actors"),
                plot =  movieInfo.getString("plot"),
                path = mediaFile.getString("path")
        )
    }

    fun jsonArrayToMovieEventList(jsonList: JSONArray): List<MovieEvent> {
        val movieList = ArrayList<MovieEvent>()
        for (i in 0 until jsonList.length()) {
            val mediaFileEvent = jsonList.getJSONObject(i)
            val mediaFile = mediaFileEvent.getJSONObject("mediaFile")

            val movie = mediaFileToMovie(mediaFile)
            movieList.add(MovieEvent(mediaFileEvent.getString("event"), movie, mediaFileEvent.getString("relativePath")))
        }
        return movieList
    }
}
